src/DrawManager.sol

```diff
@@ -209,28 +218,51 @@ contract DrawManager {
     uint24 drawId = prizePool.getDrawIdToAward(); 
     uint48 closesAt = prizePool.drawClosesAt(drawId);
     if (closesAt > block.timestamp) revert DrawHasNotClosed();
-    StartDrawAuction memory lastRequest = _lastStartDrawAuction;
-    if (lastRequest.drawId == drawId) revert AlreadyStartedDraw();
     if (rng.requestedAtBlock(_rngRequestId) != block.number) revert RngRequestNotInSameBlock();
+    
+    StartDrawAuction memory lastRequest = getLastStartDrawAuction();
+    uint auctionOpenedAt;
+    
+    if (lastRequest.drawId != drawId) { // if this request is for a new draw
+      // auctioned opened at the close of the draw
+      auctionOpenedAt = closesAt;
+      // clear out the old ones
+      while (_startDrawAuctions.length > 0) {
+        _startDrawAuctions.pop();
+      }
+    } else { // the old request is for the same draw
+      if (!rng.isRequestFailed(lastRequest.rngRequestId)) { // if the request failed
+        revert AlreadyStartedDraw();
+      } else if (_startDrawAuctions.length == maxRetries) { // if request has failed and we have retried too many times
+        revert RetryLimitReached();
+      } else if (_rngRequestId <= lastRequest.rngRequestId) { // if the request has failed, we haven't retried too many times, but the rng request has been used
+        revert StaleRngRequest();
+      } else {
+        // auctioned opened at the close of the last auction
+        // NOTE: is this accurate enough?
+        auctionOpenedAt = lastRequest.closedAt;
+      }
+    }
 
-    uint48 auctionElapsedTimeSeconds = _elapsedTimeSinceDrawClosed(block.timestamp, closesAt);
-    if (auctionElapsedTimeSeconds > auctionDuration) revert AuctionExpired();
-
-    _lastStartDrawAuction = StartDrawAuction({
+    _startDrawAuctions.push(StartDrawAuction({
       recipient: _rewardRecipient,
-      startedAt: uint40(block.timestamp),
+      closedAt: uint40(block.timestamp),
       drawId: drawId,
       rngRequestId: _rngRequestId
-    });
+    }));
 
-    (uint[] memory rewards,) = computeRewards(auctionElapsedTimeSeconds, 0, prizePool.reserve() + prizePool.pendingReserveContributions());
+    uint48 auctionElapsedTimeSeconds = _computeElapsedTime(block.timestamp, auctionOpenedAt);
+    if (auctionElapsedTimeSeconds > auctionDuration) revert AuctionExpired();
```

Comment:
We can move this above the `_startDrawAuctions.push(...)` line to revert before the storage is modified.